relay-server/src/services/processor/event.rs

```diff
@@ -72,9 +72,15 @@ pub fn extract<G: EventProcessing>(
         relay_log::trace!("processing json event");
         sample_rates = item.take_sample_rates();
         metric!(timer(RelayTimers::EventProcessingDeserialize), {
+            let (mut annotated_event, len) = event_from_json_payload(item, None)?;
             // Event items can never include transactions, so retain the event type and let
-            // inference deal with this during store normalization.
-            event_from_json_payload(item, None)?
+            // inference deal with this during normalization.
+            if let Some(event) = annotated_event.value_mut() {
+                if matches!(event.ty.value(), Some(EventType::Transaction)) {
+                    event.ty.set_value(None);
+                }
+            }
```

Comment:
Order of things happening:

1. POPs deserialization.
2. POPs normalization.
3. Processing deserialization.
4. ~Processing normalization~ (we're getting rid of this).

During deserialization:
- If we always reset, step 3 above removes the type of error events (bad).
- If we never reset, step 2 will consider the payload as a transaction and reject it because it's invalid.
- If we only reset for transactions, the first layer of relays fixes the payloads and the subsequent layers receive fixed traffic directly.